source/val/validate_annotation.cpp

```diff
@@ -266,11 +266,22 @@ spv_result_t ValidateDecorationTarget(ValidationState_t& _, SpvDecoration dec,
                << "BuiltIns can only target variables, structure members or "
                   "constants";
       }
-      if (_.HasCapability(SpvCapabilityShader) &&
-          inst->GetOperandAs<SpvBuiltIn>(2) == SpvBuiltInWorkgroupSize) {
-        if (!spvOpcodeIsConstant(target->opcode())) {
+      if (inst->GetOperandAs<SpvBuiltIn>(2) == SpvBuiltInWorkgroupSize) {
+        if (_.HasCapability(SpvCapabilityShader) &&
+            !spvOpcodeIsConstant(target->opcode())) {
           return fail(0) << "must be a constant for WorkgroupSize";
         }
+
+        uint64_t x_size, y_size, z_size;
+        bool static_x = _.GetConstantValUint64(target->word(3), &x_size);
+        bool static_y = _.GetConstantValUint64(target->word(4), &y_size);
```

Comment:
Ok so the issue is the SPIR-V spec for `WorkgroupSize` says

> See the client API specification for more detail.

and Vulkan has the [restriction](https://www.khronos.org/registry/vulkan/specs/1.3-extensions/html/vkspec.html#VUID-WorkgroupSize-WorkgroupSize-04426) it needs to be a constant (my test only had `ConstantComposite`) but I see OpenCL can also be a `OpVariable` as well

> Since this is in the validator, this code should verify instruction word count. And it should verify the type of the value paramter.

as stated above, we already validate in the builtin section that for Vulkan it is a `ConstantComposite`, will add a test and make sure `Kernel` path is checked as well

Reply:
Actually, I just realized if it is a `OpVariable` then it is not static and should can't be validated for a zero product sum, so **simpler** fix is to only validate  `WorkgroupSize` if it is a constant